Issue building with core/go/1.6.*

So I'm currently trying to build [hashicorp/consul](https://github.com/hashicorp/consul) with core/go/1.6.3. My plan file looks like the following.

<code>plan.sh</code>

``` shell
pkg_name=consul
pkg_origin=hashicorp
pkg_version=v0.7.0-rc2
pkg_description="Consul is a tool for service discovery, monitoring and configuration."
pkg_upstream_url=https://www.consul.io/
pkg_license=('mpl2')
pkg_maintainer="Jason Riddle <jallriddle@gmail.com>"
pkg_source=https://github.com/${pkg_origin}/${pkg_name}/archive/${pkg_version}.tar.gz
pkg_shasum=327c72211af1d6613477856c15bf7bcfc3c80d23542fcffe535d30d09347e08b
pkg_interpreters=(bin/bash)
pkg_deps=(core/coreutils)
pkg_build_deps=(core/go/1.6.3 core/git core/make core/which)
pkg_bin_dirs=(bin)
pkg_expose=(8300 8301 8302 8400 8500 8600)
pkg_svc_run="bin/consul agent -config-dir $pkg_svc_config_path -data-dir $pkg_svc_data_path"

do_begin() {
  export GOPATH="${HAB_CACHE_ARTIFACT_PATH}/.go"
  export GOBIN="${GOPATH}/bin"
  export HASHICORP_ORG="${GOPATH}/src/github.com/hashicorp"
  export CONSUL_REPO="${HASHICORP_ORG}/consul"
}

do_download() {
  mkdir -p ${GOPATH}
  rm -rf ${CONSUL_REPO}
  git clone https://github.com/hashicorp/${pkg_name}.git ${CONSUL_REPO} \
    --branch ${pkg_version} --single-branch
}

do_verify() {
  return 0
}

do_unpack() {
  return 0
}

do_prepare() {
  pushd ${CONSUL_REPO}
  fix_interpreter "scripts/*.sh" core/coreutils bin/env
  popd
}

do_build() {
  pushd ${CONSUL_REPO}
  make tools
  PATH="${PATH}:${GOBIN}" make dev
  popd
}

do_install() {
  pushd ${CONSUL_REPO}
  mkdir -p $pkg_prefix/bin
  cp bin/consul $pkg_prefix/bin
  popd
}
```

Upon attempting to build, I get the following error.

```
--> Running go fmt
==> Removing old directory...
==> Building...
Number of parallel builds: 1

-->     linux/amd64: github.com/hashicorp/consul

1 errors occurred:
--> linux/amd64 error: exit status 2
Stderr: # github.com/hashicorp/consul/vendor/golang.org/x/sys/unix
vendor/golang.org/x/sys/unix/asm.s:8: #include: open /hab/pkgs/core/go/1.6.3/20160728163358/pkg/include/textflag.h: no such file or directory

make: *** [GNUmakefile:19: dev] Error 1
```

I did some googling and it looked like this issue was resolved in 1.4 (https://github.com/golang/go/issues/9211) which is why I'm confused as to why this isn't able to build. Any thoughts?

--- a/go/plan.sh
+++ b/go/plan.sh
@@ -106,6 +106,9 @@ do_install() {
 
   # Remove unneeded Windows files
   rm -fv "$pkg_prefix/src/*.bat"
+
+  # Move header files to the correct place
+  cp -arv "$pkg_prefix/src/runtime" "$pkg_prefix/pkg/include"
 }
 
 do_strip() {

